extensions/amp-playbuzz/0.1/amp-playbuzz.js

```diff
@@ -0,0 +1,290 @@
+    iframe.setAttribute('allowfullscreen', 'true');
+    iframe.src = this.generateEmbedSourceUrl_();
+
+    this.listenToPlaybuzzItemMessage_('resize_height',
+      utils.debounce(this.itemHeightChanged_.bind(this), 100));
+
+    this.element.appendChild(this.getOverflowElement_());
+
+    this.applyFillContent(iframe);
+    this.element.appendChild(iframe);
+
+    return this.iframePromise_ = this.loadPromise(iframe).then(() => {
+
+      this.attemptChangeHeight(this.itemHeight_).catch(() => {/* die */});
+
+      const unlisten = this.getViewport().onChanged(
+        utils.debounce(this.sendScrollDataToItem_.bind(this), 250));
+      this.unlisteners_.push(unlisten);
+    });
+  }
+
+  /** @return {!Element} @private */
+  createPlaybuzzLoader_() {
+    const doc = this.element.ownerDocument;
+    const createElement = utils.getElementCreator(doc);
+
+    const loaderImage = createElement('img', 'pb_feed_anim_mask');
+    loaderImage.src = logo;
+
+    const loaderText = createElement('div', 'pb_feed_loading_text');
+    loaderText.textContent = 'Loading...';
+
+    const loadingPlaceholder =
+      createElement('div', 'pb_feed_placeholder_container',
+        createElement('div', 'pb_feed_placeholder_inner',
+          createElement('div', 'pb_feed_placeholder_content', [
+            createElement('div', 'pb_feed_placeholder_preloader', loaderImage),
+            loaderText,
+          ])));
+
+    return loadingPlaceholder;
+  }
+
+  /**
+   * @param {number} height
+   */
+  itemHeightChanged_(height) {
+
+    if (isNaN(height) || height === this.itemHeight_) {
+      return;
+    }
+
+    this.itemHeight_ = height; //Save new height
+
+    this.attemptChangeHeight(this.itemHeight_).catch(() => {/* die */});
+  }
+
+
+  /**
+   * @param {string} messageName
+   * @param {function} handler
+   */
+  listenToPlaybuzzItemMessage_(messageName, handler) {
+    const unlisten = events.listen(this.win, 'message',
+      event => utils.handleMessageByName(this.iframe_,
+        event, messageName, handler));
+    this.unlisteners_.push(unlisten);
+  }
+
+  /**
+   *
+   * Returns the composed embed source url
+   * @returns {string} url
+   *
+   */
+  generateEmbedSourceUrl_() {
+    const itemSrc = parseUrl(this.item_);
+    const winUrl = this.win.location;
+    const params = {
+      itemUrl: removeFragment(itemSrc.href).replace(itemSrc.protocol, ''), //remove scheme (cors) & fragment
+      relativeUrl: itemSrc.pathname,
+      displayItemInfo: this.displayItemInfo_,
+      displayShareBar: this.displayShareBar_,
+      displayComments: this.displayComments_,
+      parentUrl: removeFragment(winUrl.href),
+      parentHost: winUrl.hostname,
+    };
+
+    const embedUrl = utils.composeEmbedUrl(params);
+    return embedUrl;
+  }
+
+  sendScrollDataToItem_(changeEvent) {
+    const viewport = this.getViewport();
+
+    const scrollingData = {
+      event: 'scroll',
+      windowHeight: changeEvent.height,
+      scroll: changeEvent.top,
```

Comment:
So, to repeat question here. The intersection observer use is mostly equivalent if this code was:

```
{
  scroll: scrollTop - offsetTop,
  offsetTop: 0,
}
```

The nuance here is that you don't know the exact top scroll position of the parent document and your position within it. But you DO know exactly which part of your embed is visible. Basically this emulates a situation where your embed is always at the very top of the page (offsetTop = 0).

Can you please comment on this? I'd like to get an idea if this is sufficient information for you or if you need more?